packages/components/src/navigator/navigator-screen/component.tsx

```diff
@@ -100,11 +100,13 @@ function UnconnectedNavigatorScreen(
 		// - when the screen becomes visible
 		// - if the wrapper ref has been assigned
 		// - if focus hasn't already been restored for the current location
+		// - if the `skipFocus` option is set to `true`. This is useful when we trigger the navigation outside of NavigatorScreen.
 		if (
 			isInitialLocation ||
 			! isMatch ||
 			! wrapperRef.current ||
-			locationRef.current.hasRestoredFocus
+			locationRef.current.hasRestoredFocus ||
+			location.skipFocus
```

Comment:
let's say that we trigger a `goTo` navigation with `skipFocus = true`. At a later time, we navigate "back" (either via `goBack`, or via `goToParent`) to that same location. Should we skip focussing in that case too? Should we allow setting the the `skipFocus` specifically for those navigations too? Or should we never skip focus when navigating back?

Reply:
Probably we could, but it's not a use case right now, so I think we could leave it for now. No strong opinions, but I don't think we would need a full 'external' navigation..

Reply:
Sounds good. Let's assume, at least for now, that navigating backwards will always restore focus.